MoreLinq/RandomSubset.cs

```diff
@@ -61,40 +100,43 @@ public static IEnumerable<T> RandomSubset<T>(this IEnumerable<T> sequence, int s
             if (sequence == null) throw new ArgumentNullException(nameof(sequence));
             if (subsetSize < 0) throw new ArgumentOutOfRangeException(nameof(subsetSize));
 
-            return _(); IEnumerable<T> _()
-            {
-                // The simplest and most efficient way to return a random subet is to perform
-                // an in-place, partial Fisher-Yates shuffle of the sequence. While we could do
-                // a full shuffle, it would be wasteful in the cases where subsetSize is shorter
-                // than the length of the sequence.
-                // See: http://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle
+            return RandomSubsetImpl(sequence, rand, source => (source.ToArray(), subsetSize));
+        }
+
+        static IEnumerable<T> RandomSubsetImpl<T>(IEnumerable<T> sequence, Random rand, Func<IEnumerable<T>, (T[] array, int subsetSize)> seeder)
```

Comment:
The element names on the tuple are unused so let's skip them. This is not a public API so no one will clear. The usage is obvious from the decomposition in the implementation:

```c#
var (array, subsetSize) = seeder(sequence);
```